src/cbor/IoTCloudMessageDecoder.cpp

```diff
@@ -0,0 +1,171 @@
+/*
+  This file is part of the ArduinoIoTCloud library.
+
+  Copyright (c) 2024 Arduino SA
+
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+/******************************************************************************
+    INCLUDE
+ ******************************************************************************/
+
+#include <Arduino.h>
+
+#include "IoTCloudMessageDecoder.h"
+#include <AIoTC_Config.h>
+
+/******************************************************************************
+    PUBLIC MEMBER FUNCTIONS
+ ******************************************************************************/
+
+/******************************************************************************
+    PRIVATE MEMBER FUNCTIONS
+ ******************************************************************************/
+
+static bool copyCBORStringToArray(CborValue * param, char * dest, size_t dest_size) {
+  if (cbor_value_is_text_string(param)) {
+    // NOTE: keep in mind that _cbor_value_copy_string tries to put a \0 at the end of the string
+    if(_cbor_value_copy_string(param, dest, &dest_size, NULL) == CborNoError) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+// FIXME dest_size should be also returned, the copied byte array can have a different size from the starting one
```

Comment:
can we just store input `dest_size`, compare it with `_cbor_value_copy_string` `_dest_size` and return `false` if not matching ?